wgpu/tests/common/mod.rs

```diff
@@ -187,18 +187,28 @@ pub fn initialize_test(parameters: TestParameters, test_function: impl FnOnce(Te
 
     let missing_features = parameters.required_features - adapter_features;
     if !missing_features.is_empty() {
+        #[cfg(all(target_arch = "wasm32", feature = "webgl"))]
+        delete_html_canvas();
+
+        // TODO: we probably should use log crate here for logging also to wasm console
         println!("TEST SKIPPED: MISSING FEATURES {:?}", missing_features);
         return;
     }
 
     if !parameters.required_limits.check_limits(&adapter_limits) {
+        #[cfg(all(target_arch = "wasm32", feature = "webgl"))]
+        delete_html_canvas();
```

Comment:
Could we use a surface guard to do this such we don't need to remember it for every branch?

Reply:
Yeah, that would be great. Do you mean something like the C++ RAII pattern? How would you do that?

Reply:
I'd have create_html_canvas return a type which impls Drop that destroys the canvas

Reply:
It works. Thanks for suggesting the idea!